### PR TITLE
feat(admin api): new /refresh api endpoint

### PR DESCRIPTION
**Pull request by authcompanion**
Simplify the ability to programmatically integrate with AuthC via an Admin. Today the Admin API is tightly coupled to the AuthC provided admin ui; relax this. 

Scope of work:

## Functionality
- [x] Add Admin API token refresh route to refresh the admin JWT
- [x] On the Admin API remove the fingerprint mechanism required to authenticate the admin JWT when using Admin APIs. Keep the mechanism for security when using the public auth apis. 
- [x] When retrieving an JWT from the Admin API, also return a Refresh Token in the payload for use in third party clients, continue to have the server set the refresh token as a cookie  for /login (needed for admin dashboard)

## Documentation:
- [x] Create a doc to document all the default claims for JWT token generation across access and refresh tokens. 
- [ ] Create doc on how to include a reverse proxy to route to different admin port and auth port. 
- [x] Add doc for the new refresh endpoint on admin api

## Wrap up
- [x] Bug fix, stabilize, improve error msgs

### CODE REVIEW

**Comment by Teddy-Schmitz** [comment]
i think you need to fixup your prettifier config, lots of style changes in this PR

**Comment by authcompanion** [comment]
> i think you need to fixup your prettifier config, lots of style changes in this PR

Agreed. Adding a prettier config so we don't have to deal with different formatting for our PRs.

File: routes/admin.routes.js [hunk 1]
```diff
@@ -49,6 +52,11 @@ const adminRoutes = async function (fastify, options) {
     { onRequest: [authenticateAdminRequest] },
     logoutHandler
   );
+  fastify.post(
+    "/refresh",
+    { onRequest: [authenticateAdminRequest], ...refreshSchema },
```
[the remark this answers is not in the document]
Comment (authcompanion) [reply routes/admin.routes.js]: fixed thnx

File: plugins/db/schema/2__admin.sql [hunk 2]
```diff
@@ -6,7 +6,7 @@ CREATE TABLE admin (
     email TEXT NOT NULL UNIQUE,
     password TEXT NOT NULL,
     challenge TEXT,
-    jwt_id TEXT NOT NULL,
+    jwt_id TEXT,
```
[the remark this answers is not in the document]
Comment (authcompanion) [reply plugins/db/schema/2__admin.sql:9]: SQLite no support for `alter column` to change the column data type to remove not null. https://stackoverflow.com/questions/4007014/alter-column-in-sqlite/4820137#4820137 - reverting this change so as not to introduce this breaking change.

[the remark this answers is not in the document]
Comment (authcompanion) [reply plugins/db/schema/2__admin.sql]: Thnx for calling this out and the sanity check.

**Pull request merged**